### PR TITLE
feat(debug): add chunk endorsement ratio to last_blocks debug page

### PR DESCRIPTION
**Pull request by Trisfald**
Adds a new metric in the `last_blocks` debug page: `endorsement(stake)`. This indicates what percentage of the 'validators stake' has endorsed a given chunk in a block. 

The addition works only if the stateless validation feature is enabled.

### CODE REVIEW

**Comment by Trisfald** [comment]
The following is an example when a node (node3 in this case) exits the network and can't validate chunks anymore:

chunk_endorsements

**Review by wacban — APPROVED** [review]
LGTM, thank you!

File: chain/client/src/debug.rs [hunk 1]
```diff
@@ -625,7 +633,89 @@ impl ClientActorInner {
...
+    fn compute_chunk_endorsements_ratio(
+        &self,
+        block: &Option<Block>,
+    ) -> Option<HashMap<ChunkHash, f64>> {
+        match block {
+            Some(block) => {
+                let mut chunk_endorsements = HashMap::new();
+                if block.chunks().len() != block.chunk_endorsements().len() {
+                    return None;
+                }
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply chain/client/src/debug.rs:648]: It should not be possible in fact, if this happens the block will be rejected in `preprocess_block`. Not sure then if the block can show up in the debug page at all

[the remark this answers is not in the document]
Comment (wacban) [reply chain/client/src/debug.rs:648]: In that case we don't need to worry about it, those blocks would not be shown in the debug page.

File: chain/client/src/debug.rs [hunk 2]
```diff
@@ -464,6 +467,11 @@ impl ClientActorInner {
                                 chunk.chunk_hash().0,
                             )
                             .map(|s| s.whole_milliseconds() as u64),
+                            endorsement_ratio: chunk_endorsements
+                                .as_ref()
+                                .map(|chunks| chunks.get(&chunk.chunk_hash()))
+                                .flatten()
+                                .copied(),
```
[the remark this answers is not in the document]
Comment (Trisfald) [reply chain/client/src/debug.rs]: Done!

**Pull request merged**